### PR TITLE
Add missing API documentation for the crypto package

### PR DESCRIPTION
**Pull request by Liamsi**
part of #32

### CODE REVIEW

File: crypto/sign/sign.go [hunk 1]
```diff
@@ -9,18 +9,35 @@ import (
...
-	SignatureSize  = 64
+	// PublicKeySize is the size of the public-key in bytes.
+	PublicKeySize = 32
+	// SignatureSize is the size of the created signature in bytes.
+	SignatureSize = 64
 )
 
 var (
+	// ErrorGetPubKey will be returned if the public-key can't be derived from the underlying private-key.
+	// FIXME This is never used ... I think it can be deleted (PrivateKey.Public() already indicates if this happened.
```
Comment (c633) [inline crypto/sign/sign.go]: Yes, if it makes sense to you.

File: crypto/util.go [hunk 2]
```diff
@@ -48,6 +63,8 @@ func NewCommit(stuff ...[]byte) (*Commit, error) {
 	}, nil
 }
 
+// Verify verifies that the underlying commit `c` was a commit to the passed
+// data `stuff` (which wont be mutated).
```
Comment (c633) [inline crypto/util.go]: s/wont/won't

**Comment by c633** [comment]
Generally this pull LGTM.

File: crypto/doc.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// Package crypto contains some cryptographic routines, to:
+// - hash arbitrary data (`Digest`)
+// - create a cryptographic commit to arbitrary data
+// - generate a random slice of bytes
```
Comment (masomel) [inline crypto/doc.go]: Should we mention this slice is hashed to hide the PRG output?

File: crypto/sign/sign.go [hunk 4]
```diff
@@ -2,25 +2,35 @@ package sign
...
 import (
 	"crypto/rand"
-	"errors"
 	"io"
 
 	"golang.org/x/crypto/ed25519"
 )
 
 const (
+	// PrivateKeySize is the size of private-key in bytes.
```
Comment (masomel) [inline crypto/sign/sign.go]: s/private-key/the private-key (for consistency with the other consts).

File: crypto/sign/sign.go [hunk 5]
```diff
@@ -29,15 +39,24 @@ func GenerateKey(rnd io.Reader) (PrivateKey, error) {
...
 
+// Sign returns a signature on the passed byte-slice `message` using the
+// underlying private-key.
+// The passed slice won't be modified.
 func (key PrivateKey) Sign(message []byte) []byte {
 	return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
+// Public derives the corresponding public-key from the underlying private-key.
+// It returns the derived key ,if possible, and a boolean flag which indicates if
```
Comment (masomel) [inline crypto/sign/sign.go]: Super nit-picky: s/key ,if/key, if

File: crypto/util.go [hunk 6]
```diff
@@ -8,10 +8,14 @@ import (
...
 
 const (
+	// HashSizeByte is the used hash's value size in bytes.
 	HashSizeByte = 32
-	HashID       = "SHAKE128"
+	// HashID identifies the used hash as a string.
+	HashID = "SHAKE128"
 )
 
+// Digest hashes all passed byte slice and hashes them.
```
Comment (masomel) [inline crypto/util.go]: s/slice/slices

File: crypto/doc.go [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+// - sign data and verify signatures
```
Comment (masomel) [inline crypto/doc.go]: Should we mention the supported signature algorithms?

File: crypto/doc.go [hunk 8]
```diff
@@ -0,0 +1,7 @@
+// Package crypto contains some cryptographic routines, to:
+// - hash arbitrary data (`Digest`)
```
Comment (masomel) [inline crypto/doc.go]: Should we mention the supported hash algorithms?

File: crypto/util.go [hunk 9]
```diff
@@ -22,7 +26,10 @@ func Digest(ms ...[]byte) []byte {
 	return ret
 }
 
-// MakeRand generates a random slice of byte and hashes it.
+// MakeRand returns a random slice of byte.
+// It returns an error if there was a problem while generating the random slice.
+// It is different from the 'standard' random byte generation as it hashes its
+// output before returning it.
```
Comment (masomel) [inline crypto/util.go]: Should this explain (briefly) the reason for hashing the output?

**Comment by masomel** [comment]
LGTM

**Pull request merged**